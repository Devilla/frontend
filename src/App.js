import React from 'react';
import { browserHistory } from 'react-router';
import { AppContainer } from 'react-hot-loader';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import rootSaga from './sagas';
import reducer from './ducks';
import thunk from 'redux-thunk';
import { syncHistoryWithStore } from 'react-router-redux';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import createSagaMiddleware from 'redux-saga';
import { StripeProvider } from 'react-stripe-elements';

const sagaMiddleware = createSagaMiddleware();

export const store = createStore(reducer, composeWithDevTools(
  applyMiddleware(sagaMiddleware, thunk)
));

const routerHistory = syncHistoryWithStore(browserHistory, store, {selectLocationState: state => state.get('router')});


sagaMiddleware.run(rootSaga);

const App = ({Component}) =>
  <AppContainer errorReporter={Error}>
    <Provider store={store}>
      <StripeProvider apiKey={process.env.NODE_ENV === 'production' ? process.env.REACT_APP_STRIPE_KEY : process.env.REACT_APP_DEVELOPMENT_STRIPE_KEY}>
        <Component routerHistory={routerHistory} />
      </StripeProvider>
    </Provider>
  </AppContainer>;

export default App;
