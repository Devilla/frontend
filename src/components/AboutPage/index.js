import React, { Component } from 'react';
import {Animated} from "react-animated-css";
export default class About extends Component {
     componentDidMount(){
     window.scrollTo(0, 0)
   } 
  render() {
    return (
        <div>
            <div className="page-header"><div className="page-header-overlay">
            <div className="centered page-header-container w-container">
            <Animated className="page-header-title" animationIn="fadeInDown" animationOut="fadeOut" isVisible={true}>About us</Animated>
            </div></div></div>
            <div className="section innerpage">
                <div className="container w-container">
                    <p>We started Influence with one mission: to cut down the advertising cost and to convert more customers on your website.</p>
            <p>We use your existing customers’ data on your website and show it to your visitors so that they are socially influenced by buying behavior of people on the website.</p>
            <p>Using this tool we’ve been able to help a lot of businesses in increasing their money making and get more conversions.</p>
            <p>Social influence is the next big thing in marketing for influencing buyers to buy more.</p>
            <p>Let’s start with your free to start trial and get making more money for you.
            </p>
                   
                </div>
              </div>
        </div>
    );
  }
}
