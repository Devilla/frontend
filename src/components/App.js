import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import  '../assets/stylesheets/slick.css'
import  '../assets/stylesheets/slick-theme.css'
import  '../assets/stylesheets/style.css'

import Header from './_common/Header';
import Footer from './_common/Footer';
import { Spinner } from './index.js';

class App extends Component {
  render(){
    return (
        <div className="app-container">
          <Header />

          <Spinner loading={this.props.loading} />
          <div className="content">
            {this.props.children}
          </div>
          <Footer />
        </div>
    );
  }
};

App.propTypes = {
  children: PropTypes.node
};

const mapStateToProps = state => ({
  loading: state.getIn(['loading', 'state'])
});

export default connect(mapStateToProps)(App);
