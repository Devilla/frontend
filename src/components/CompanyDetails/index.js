import React, { Component } from 'react';
import { Animated } from "react-animated-css";
import { ToastContainer, toast } from 'react-toastify';
import { validateCompanyName, validateWebsite, submitCompanyDetails } from '../../services/FormUtils';
import { storeToken } from '../../services/Request';


const toastConfig = {
  position: toast.POSITION.BOTTOM_LEFT,
  autoClose: 2000
};

export default class CompanyDetails extends Component {
  constructor(props) {
    super(props);

    this.state = {
      companyName: '',
      website: ''
    };

    // Store the token passed in params. This will be used to continue process further
    storeToken(props.match.params.token);
  }

  // Allow empty field
  isCompanyNameValid = () => (this.state.companyName === '' || validateCompanyName(this.state.companyName));

  // Allow empty field
  isWebsiteValid = () => (this.state.website === '' || validateWebsite(this.state.website));

  handleInputChange = event => {
    const { name, value } = event.target;
    this.setState({ [name]: value});
  };

  handleCompanyNameBlur = event => {
    if(!this.isCompanyNameValid())
      toast.error("Please enter company name", toastConfig);
  };

  handleWebsiteBlur = event => {
    if(!this.isWebsiteValid())
      toast.error("Please enter a valid website url", toastConfig);
  };

  handleSubmit = event => {
    event.stopPropagation();
    event.preventDefault();

    submitCompanyDetails(this.state.companyName, this.state.website).then(res => {
      console.log(res);
      // TODO: check response before treating it as successfull
      toast.info('Successfull', toastConfig);
    }).catch(err => {
      console.log(err);
      toast.error('Something went wrong!', toastConfig);
    });
  };

  render() {
    const isFormValid = this.isCompanyNameValid() && this.isWebsiteValid();
    console.log(this.state.website, this.isWebsiteValid());
    return (
      <div className="authpage section innerpage">
        <div className="container">
          <div className="wrapper">
            <Animated className="leftwrap center" animationIn="fadeIn" animationOut="fadeOut" isVisible={true}>
              <form className="loginfrm" onSubmit={this.handleSubmit} >
                <h3 className="dashed">Tell us about your company</h3>
                <div className="section-divider-line"></div>
                <div className="frmcntl">
                  <input className="field w-input"
                    name="companyName"
                    value={this.state.companyName}
                    onBlur={this.handleCompanyNameBlur}
                    onChange={this.handleInputChange}
                    placeholder="Company Name"
                  />
                </div>
                <div className="frmcntl">
                  <input
                    className="field w-input"
                    name="website"
                    placeholder="Website"
                    value={this.state.website}
                    onBlur={this.handleWebsiteBlur}
                    onChange={this.handleInputChange}
                  />
                </div>
                <div className="frmcntl">
                  <input className="button submit-button w-button"
                    type="submit"
                    value="Next"
                    disabled={!isFormValid}
                  />
                </div>

                <ToastContainer hideProgressBar={true} />
              </form>
              <div className="support"></div>
            </Animated>
          </div>
        </div>
      </div>
    );
  }
}