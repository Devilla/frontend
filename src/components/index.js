export { default as WebsiteHeader } from "./WebsiteHeader/WebsiteHeader";
export { default as WebsiteFooter } from "./WebsiteFooter/WebsiteFooter";
export { default as WebsiteHome } from "./WebsiteHome/WebsiteHome";
export { default as WebsiteHowItWorks } from "./WebsiteHowItWorks/WebsiteHowItWorks";
export { default as WebsiteIntegrations } from "./WebsiteIntegrations/WebsiteIntegrations";
export { default as WebsitePricing } from "./WebsitePricing/WebsitePricing";
export { default as WebsiteAbout } from "./WebsiteAbout/WebsiteAbout";
export { default as WebsiteSignIn } from "./WebsiteSignIn/WebsiteSignIn";
export { default as WebsiteSignUp } from "./WebsiteSignUp/WebsiteSignUp";
export { default as WebsiteContact } from "./WebsiteContact/WebsiteContact";
export { default as WebsiteTerms } from "./WebsiteTerms/WebsiteTerms";
export { default as WebsitePolicy } from "./WebsitePolicy/WebsitePolicy";
export { default as WebsiteDemoPage}  from "./WebsiteDemoPage/WebsiteDemoPage";
export { default as WebsiteCustStory}  from "./WebsiteCustStory/WebsiteCustStory";
export { default as WebsiteFeature}  from "./WebsiteFeature/WebsiteFeature";
export { default as Affiliate}  from "./Affiliate/Affiliate";
export { default as AffiliateRegister}  from "./Affiliate/AffiliateRegister";

export { default as Header } from "./Header/Header";
export { default as ForgetPassword } from "./ForgetPassword";
export { default as PricePage } from "./PricePage";
export { default as Common } from "./Common/function";
export { default as ResetPassword } from "./ResetPassword";
export { default as Sidebar } from "./Sidebar/Sidebar";
export { default as StatsCard } from "./StatsCard/StatsCard";
export { default as Dashboard } from './Dashboard/Dashboard';
export { default as Notification } from './Notification';
export { default as Analytics } from './Analytics/analytics';
export { default as AnalyticsProfile } from './Analytics/AnalyticsProfile';
export { default as New } from './New/new';
export { default as LoginFlow } from './LoginFlow/LoginFlow';
export { default as Spinner } from './Spinner/Spinner';
export { default as Error } from './Error/Error';
export { default as ComingSoon } from './ComingSoon/ComingSoon';
export { default as Button } from './Button/Button';
export { default as Profile } from './Profile/Profile';
export { default as BillingDetails } from './Billing/BillingDetails';
export { default as UpgradeCard } from './UpgradeCard/UpgradeCard';
export { default as UpgradePlan } from './UpgradePlan/UpgradePlan';
export { default as Integration } from './Integration/Integration';
export { default as Template } from './Template';
export { default as VerificationPage } from './VerificationPage/VerificationPage';
