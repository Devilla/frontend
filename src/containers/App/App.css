@import url('https://fonts.googleapis.com/css?family=Open+Sans:200,300,400,400i,500,600,700');
@import url('https://fonts.googleapis.com/css?family=Source+Sans+Pro:200,300,400,400i,600,700');

body {
  font-family: "Raleway, Source Sans Pro, Verdana, Helvetica", sans-serif;
}
