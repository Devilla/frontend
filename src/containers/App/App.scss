@import url('https://fonts.googleapis.com/css?family=Open+Sans:200,300,400,400i,500,600,700');
@import url('https://fonts.googleapis.com/css?family=Source+Sans+Pro:200,300,400,400i,600,700');

body {
  font-family: "Raleway, Source Sans Pro, Verdana, Helvetica", sans-serif;
  display: block;
  margin: 0px;
}

.website-app {
  .alert-warning {
    background-color: #f2dede;
    border-color: #ebccd1;
    color: #a94442;
  }
  .content {
    padding-top: 100px;
  }
  .toast-style {
    padding: 17px 7px;
    background-color: rgba(0, 0, 0, 0.9) !important;
    color: #fff !important;
    text-align: center !important;
    height: 10px !important;
    border-radius: 5px !important;
    button {
      height: 1.5em;
      color: #fff;
    }
  }

  .transition-wrapper {
    position: relative;
    z-index: 1;
    .transition-item {
      position: fixed;
      top: 0;
      right: 0;
      left: 0;
    }
  }
  .signin-container,
  .authpage,
  .websitepolicy-container,
  .websitehowitworks-container,
  .websitehome-container,
  .WebsiteTerms-container,
  .websitepricing-container,
  .website-payment-container,
  .websiteintegrations-container,
  .demopage-container,
  .website-cust-story,
  .websitecontact-container,
  .websiteabout-container,
  .WebsiteFeature-container,
  .affiliateregister-container
   {
    overflow: hidden;
    box-sizing: border-box;
    padding: 20px;
    min-height: 100vh;
    transition: 500ms ease-out;

    &.transition-leave {
      opacity: 1;
    }

    &.transition-leave.transition-leave-active {
      opacity: 1;
      transform: translate3d(100%, 0, 0);
    }
  }
}
