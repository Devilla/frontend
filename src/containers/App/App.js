import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ToastContainer } from 'react-toastify';
import { browserHistory } from 'react-router';
import { connect } from 'react-redux';
import { Spinner } from '../../components/index.js';
import { WebsiteHeader, WebsiteFooter } from 'components';

import { checkTokenExists } from 'ducks/auth';

import './scss/stack-interface.scss';
import './scss/socicon.css';
import './scss/iconsmind.scss';
import './scss/bootstrap.scss';
import './scss/theme.scss';
import './scss/custom.scss';
import './scss/font-raleway.scss';
import './scss/flickity.scss';
import './scss/font-sourcesanspro.scss';
import './App.scss';

class App extends Component {
  componentWillMount() {
    this.checkLogin((err) => {
      if(!err) {
        browserHistory.push('/dashboard');
      }
    });
  }

  checkLogin(callback) {
    const cookie = localStorage.getItem('authToken');
    const authToken = cookie
      ? JSON.parse(cookie)
      : null;
    if (authToken) {
      this.props.checkTokenExists(authToken);
      callback()
    } else {
      callback("not logged in")
    }
  }

  render(){
    return (
      <div className="website-app">
        <div className="basic-gradient-light" data-smooth-scroll-offset="77">
           <WebsiteHeader />
           <Spinner loading={this.props.loading} />
           <div className="content">
             {this.props.children}
           </div>
           <WebsiteFooter />
        </div>
        <ToastContainer hideProgressBar={true} />
      </div>
    );
  }

}

const mapDispatchToProps = {
  checkTokenExists
};

export default connect(null, mapDispatchToProps)(App);
