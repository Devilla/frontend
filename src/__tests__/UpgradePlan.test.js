import React from 'react'
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import renderer from 'react-test-renderer'
import { UpgradePlan } from 'components';
import { store } from 'App';


// Snapshot for website feature React Component
describe('>>>U P G R A D E  P L A N  --- Snapshot',()=>{
    it('+++capturing Snapshot of upgrade plan', () => {
        const renderedValue =  renderer.create(<UpgradePlan store={store} />).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });

});
