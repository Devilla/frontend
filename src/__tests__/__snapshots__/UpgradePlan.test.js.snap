// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>U P G R A D E  P L A N  --- Snapshot +++capturing Snapshot of upgrade plan 1`] = `
<div
  className="upgrade-plan-container"
>
  <div
    className="planlist-container"
  >
    <div
      className="content fill upgrade-plan "
    >
      <div
        className="container-fluid"
      >
        <div
          className="inlineclr row"
        >
          <div
            className="col-md-12"
          >
            <div
              className="card-box pt-0 pl-0"
            >
              <h4
                className="header-title text-left"
              >
                Upgrade your Plan
              </h4>
              <hr />
              <div
                className="plans float-right"
              >
                <div
                  className="row"
                >
                  <div
                    className="animated fadeIn leftwrap center text-center"
                    style={
                      Object {
                        "animationDelay": "0ms",
                        "pointerEvents": "all",
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "width": "100%",
                        }
                      }
                    >
                      <div
                        style={
                          Object {
                            "width": "100%",
                          }
                        }
                      >
                        <div
                          className="price pt-2"
                        >
                          <div
                            className="pricing-row w-row"
                            style={
                              Object {
                                "width": "100%",
                              }
                            }
                          >
                            <div
                              className="w-col"
                            >
                              <span
                                className="btn btn-outline-info waves-light waves-effect mr-2 set-br"
                                onClick={[Function]}
                              >
                                Monthly
                              </span>
                            </div>
                          </div>
                        </div>
                        <div
                          className="row"
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                        />
                        <div
                          className="price"
                        >
                          <div
                            className="pricing-row w-row"
                            style={
                              Object {
                                "width": "100%",
                              }
                            }
                          >
                            <div
                              className="w-col"
                            >
                              <span
                                className="active btn btn-outline-success waves-light waves-effect ml-2 set-br"
                                onClick={[Function]}
                              >
                                 Yearly 
                              </span>
                            </div>
                          </div>
                        </div>
                        <div
                          className="row"
                          style={
                            Object {
                              "opacity": 0.5,
                            }
                          }
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="float-right row"
                >
                  <div
                    className="m-t-50 "
                  >
                    <span
                      className="btn btn-primary  waves-light waves-effect number ml-2 pl-4 pr-4"
                      onClick={[Function]}
                      type="button"
                    >
                      Proceed 
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="clearfix"
      />
    </div>
  </div>
  <div />
</div>
`;
